phd-tests/framework/src/serial/raw_buffer.rs

```diff
@@ -82,62 +106,37 @@ impl Inner {
     }
 }
 
-impl Drop for Inner {
-    fn drop(&mut self) {
-        if let Err(e) = self.log.flush() {
-            error!(%e, "failed to flush serial console log during drop");
-        }
-    }
-}
-
-impl vte::Perform for Inner {
-    fn print(&mut self, c: char) {
-        self.push_character(c);
-    }
-
-    fn execute(&mut self, byte: u8) {
-        if byte == b'\n' {
-            self.push_character('\n');
-        }
-    }
-}
-
-/// A "raw" serial console buffer that handles incoming characters and newline
-/// control bytes and nothing else.
-pub(super) struct RawBuffer {
-    inner: Inner,
-    parser: vte::Parser,
-}
-
-impl RawBuffer {
-    /// Constructs a new buffer.
-    pub(super) fn new(log_path: Utf8PathBuf) -> Result<Self> {
-        let log_file = std::fs::File::create(&log_path).with_context(|| {
-            format!("opening serial console log file {}", log_path)
-        })?;
-        let writer = BufWriter::new(log_file);
-        let inner = Inner {
-            log: writer,
-            line_buffer: String::new(),
-            wait_buffer: String::new(),
-            waiter: None,
-        };
-        Ok(Self { inner, parser: vte::Parser::new() })
-    }
-}
-
 impl Buffer for RawBuffer {
     fn process_bytes(&mut self, bytes: &[u8]) {
-        for b in bytes {
-            self.parser.advance(&mut self.inner, *b);
+        use termwiz::escape::{Action, ControlCode};
+        let actions = self.parser.parse_as_vec(bytes);
+        for action in actions {
+            match action {
+                Action::Print(c) => self.push_character(c),
+                Action::PrintString(s) => {
+                    self.push_str(&s);
+                }
+                Action::Control(ControlCode::LineFeed) => {
+                    self.push_character('\n')
+                }
+                _ => {}
```

Comment:
nit, not a big deal: do you think it's worthwhile to log other control characters, instead of eating them?  i could be convinced either way.